src/util.ts

```diff
@@ -89,9 +89,32 @@ export function throttleAfter<T extends (this: any, ...args: any[]) => void>(cal
 		let context = this;
 		let args: IArguments | null = arguments;
 
-		setTimeout(function () {
+		guaranteeMinimumTimeout(function () {
 			callback.apply(context, args);
 			args = context = ran = null;
 		}, delay);
 	};
 }
+
+export function guaranteeMinimumTimeout(callback: (...args: any[]) => void, delay?: number): Handle {
+	const startTime = Date.now();
+	let timerId: number | null;
+
+	function timeoutHandler() {
+		const delta = Date.now() - startTime;
+		if (delay == null || delta >= delay) {
+			callback();
+		} else {
+			// Cast setTimeout return value to fix TypeScript parsing bug.  Without it,
+			// it thinks we are using the Node version of setTimeout.
+			timerId = <any> setTimeout(timeoutHandler, delta);
```

Comment:
Should we be timing our for `delay - delta`? What we're saying here is that we want to wait as long as we've waited right? That seems arbitrary

Reply:
Fixing now...